src/devices/Seesaw/SeesawAdc.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.Seesaw
+{
+    public partial class Seesaw : IDisposable
+    {
+        /// <summary>
+        /// Reads the value of an analog pin.
+        /// </summary>
+        /// <param name="pin">The pin number in the devices numbering scheme.</param>
+        /// <returns>A value between 0..1023 that represents the analog value.</returns> 
+        public UInt16 AnalogRead(byte pin)
+        {
+            const int AdcConversionDelay = 500; //500 uSeconds delay between read and write to allow Adc conversion time.
+
+            byte[] buffer;
+
+            if (!HasModule(Seesaw_Module.Gpio))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw ADC functionality");
+            }
+
+            if (pin < 2 || pin > 5)
+            {
+                throw new ArgumentException("ADC pin must be within 2-5 range.");
+            }
+
+            buffer = Read(Seesaw_Module.Adc, Seesaw_Function.Adc_Channel_Offset + pin - 2, 2, AdcConversionDelay);
+
+            return((UInt16)(((UInt16) buffer[0] << 8) | buffer[1]));
```

Comment:
Use one of these for conversion https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1